src/vm_pages.rs

```diff
@@ -1455,7 +1455,7 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         self.do_get_converted_pages::<Page<ConvertedDirty>>(page_addr, page_size, num_pages)
     }
 
-    /// Acquries an exclusive reference to the `num_pages` shared pages starting at `page_addr`.
+    /// Acquries an exclusive reference to the `num_pages` shared host pages starting at `page_addr`.
     pub fn get_shareable_pages(
```

Comment:
Now that I'm looking at this again, the logic in the closure passed to `get_mapped_pages()` doesn't seem right. We're checking that the entire range is physically contiguous, but really we should only care that sub-pages within the huge page are physically contiguous. `do_get_converted_pages()` has a similar problem.

cc @sboeuf

Reply:
Ah I see, let me have a look!

Reply:
#316 should cover this.